### PR TITLE
Add and Remove Note from Group 

### PR DESCRIPTION
**Pull request by maxime-dodin**
#### Description

Note basic implementation, containing Add and Remove note 

#42, #40 

#### Changelog

- [ENHANCEMENT]
Note Models and Validator, with Add and Remove request

### CODE REVIEW

File: models/mongo/notes.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+	"github.com/google/uuid"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"go.uber.org/zap"
+	"golang.org/x/net/context"
+)
+
+type notesRepository struct {
```
Comment (rojas-diego) [inline models/mongo/notes.go:16]: Same everywhere I guess. `notes` -> `groupNotes`. Let me know if you think this is unnecessary

File: models/mongo/notes.go [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+	rep := &notesRepository{
+		logger: logger.Named("mongo").Named("notes"),
+		db:     db,
+		coll:   db.Collection("notes"),
+	}
+
+	_, err := rep.coll.Indexes().CreateOne(
+		context.Background(),
+		mongo.IndexModel{
+			Keys:    bson.D{{Key: "note_id", Value: 1}, {Key: "group_id", Value: 1}},
```
Comment (rojas-diego) [inline models/mongo/notes.go:32]: What do you think about this change? This makes sure that the same note isn't assigned to two different groups. If a student wants to share one note between multiple groups he will need to make a copy.
```suggestion
			Keys:    bson.D{{Key: "note_id", Value: 1}},
```

File: models/notes.go [hunk 3]
```diff
@@ -0,0 +1,42 @@
+package models
+
+import (
+	"context"
+	"time"
+)
+
+type Note struct {
```
Comment (rojas-diego) [inline models/notes.go:8]: Same here `Note` -> `GroupNote`

**Review by rojas-diego — COMMENTED** [review]
Amazing!

File: notes.go [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+	"errors"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+func (srv *groupsAPI) AddGroupNote(ctx context.Context, in *accountsv1.AddGroupNoteRequest) (*accountsv1.AddGroupNoteResponse, error) {
+	err := validators.ValidateAddGroupNote(in)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, "failed to validate add member request")
```
Comment (rojas-diego) [inline notes.go:17]: I think you forgot to update the log message!

File: notes.go [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+		return nil, status.Error(codes.Unauthenticated, err.Error())
+	}
+
+	accountId := token.UserID.String()
+
+	_, err = srv.memberRepo.Get(ctx, &models.MemberFilter{AccountID: &accountId, GroupID: &in.GroupId})
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, "failed to get member from group_id")
+	}
+
```
Comment (rojas-diego) [inline notes.go:31]: Maybe we should check that the note exists within the notes-service? And that the user is the author of said note.

Or **Option 2** which I like better: We have a `POST /notes` endpoint on the api gateway which takes care of calling `CreateNote` in the notes-service and then calls `AddGroupNote` with the said note. And therefore `AddGroupNote` would become an internal endpoint which is only called by the API gateway. LMKWYT.

**Review by rojas-diego — COMMENTED** [review]
Sorry I missed some things.

File: notes.go (hunk 5 above)
Comment (maxime-dodin) [reply notes.go:31]: Yes, i had the second option in mind

**Review by rojas-diego — DISMISSED** [review]

**Pull request merged**